src/app/components/Countdown.tsx

```diff
@@ -0,0 +1,65 @@
+"use client"
+
+import classNames from "classnames"
+import dynamic from "next/dynamic"
+import { useEffect, useState } from "react"
+import { RightArrow } from "./RightArrow"
+
+type CountdownProps = {
+  className?: string
+  targetDate: string
+}
+
+const getDelta = (targetDate: string) => {
+  const now = new Date().valueOf()
+  const target = new Date(targetDate).valueOf()
+  const delta = target - now
+  const totalSeconds = Math.floor(delta / 1_000)
+  const seconds = totalSeconds % 60
+  const totalMinutes = Math.floor(totalSeconds / 60)
+  const minutes = totalMinutes % 60
+  const totalHours = Math.floor(totalMinutes / 60)
+  const hours = totalHours % 24
+  const days = Math.floor(totalHours / 24)
+
+  return { days, hours, minutes, seconds }
+}
```

Comment:
Maybe use https://day.js.org/ with relative plugin to achieve that simply. But it's still fine doing it like that, just worrying about being error prone.

Reply:
This seems to be harder because the relative function doesn't accept a format. there's also diff, which can be used with duration, but that seems to compute months from days which we don't want. I'll stick with this for now, if that's ok